internal/controller/pkg/revision/hook.go

```diff
@@ -182,22 +199,12 @@ func (h *ProviderHooks) Deactivate(ctx context.Context, pr v1.PackageRevision) e
 	if err := h.client.Delete(ctx, &corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: pr.GetName(), Namespace: h.namespace}}); resource.IgnoreNotFound(err) != nil {
 		return errors.Wrap(err, errDeleteProviderSA)
 	}
-	// Delete the service if it exists.
+
+	// TODO(phisco): only added to cleanup the service we were previously
+	// 	deploying for each provider revision, remove in a future release.
 	if err := h.client.Delete(ctx, &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: pr.GetName(), Namespace: h.namespace}}); resource.IgnoreNotFound(err) != nil {
 		return errors.Wrap(err, errDeleteProviderService)
 	}
-	// Delete the TLS Server Secret if it exists.
```

Comment:
Would you mind adding a note as a code comment on why we don't need to delete TLS secrets (and service?) during deactivation?

Reply:
done: https://github.com/crossplane/crossplane/pull/4656/commits/3e8db9d7e82b5f81faa03661126a37092f15e7cb